mayastor/src/core/reactor.rs

```diff
@@ -455,16 +460,19 @@ impl Reactor {
             t.poll();
         });
 
-        while let Ok(i) = self.incoming.pop() {
-            self.threads.borrow_mut().push_back(i);
+        if self.has_incoming.get() {
```

Comment:
I think this could be simplified into something like if self.has_incoming.replace(false) == true { ... }

Reply:
Done, thanks!